multi/imxrt-multi/cm4.c

```diff
@@ -233,6 +275,10 @@ static ssize_t chanWrite(id_t id, const void *buff, size_t bufflen)
 		setTXirq(id, 1);
 	}
 
+	if (i == 0 && m4_common.channel[id].state.nonblock == 1) {
+		i = -EWOULDBLOCK;
```

Comment:
On some systems `EWOULDBLOCK != EAGAIN`, and `EWOULDBLOCK` is used for sockets. In phoenix-rtos [EAGAIN is defined first](https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/master/include/errno.h#L30), and [EWOULDBLOCK is an alias](https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/master/include/errno.h#L59). Quoting the [Open Group](https://pubs.opengroup.org/onlinepubs/009604599/functions/read.html):

normal fd:
![2023-09-15-102159_789x44_scrot](https://github.com/phoenix-rtos/phoenix-rtos-devices/assets/141153/18b95802-952d-4e5a-baa3-a223e8695b0a)
pipe:
![2023-09-15-102144_955x77_scrot](https://github.com/phoenix-rtos/phoenix-rtos-devices/assets/141153/15926e90-38d5-4248-afc4-b4faa940b930)
socket:
![2023-09-15-102016_1096x122_scrot](https://github.com/phoenix-rtos/phoenix-rtos-devices/assets/141153/86b7833e-4ee0-41af-b9bf-a69b190052cf)